frame/elections-phragmen/src/lib.rs

```diff
@@ -471,19 +507,46 @@ decl_module! {
 		/// Writes: O(do_phragmen)
 		/// # </weight>
 		#[weight = (2_000_000_000, DispatchClass::Operational)]
-		fn remove_member(origin, who: <T::Lookup as StaticLookup>::Source) -> DispatchResult {
+		fn remove_member(
+			origin,
+			who: <T::Lookup as StaticLookup>::Source,
+			has_replacement: bool,
+		) -> DispatchResultWithPostInfo {
 			ensure_root(origin)?;
 			let who = T::Lookup::lookup(who)?;
 
+			let will_have_replacement = <RunnersUp<T>>::decode_len().unwrap_or(0) > 0;
+			match (will_have_replacement, has_replacement) {
+				(true, true) | (false, false) => {
+					// correct prediction. nothing
+				}
+				(false, true) => {
+					// prediction was that we will have a replacement, so we don't charge a whole
+					// lot of weight. just abort.
+					return Err(Error::<T>::InvalidReplacement.into());
+				}
+				(true, false) => {
+					// prediction was that we will NOT have a replacement, and now this call is
+					// aborting whilst charging a metric ton of weight. Refund and abort.
+					// TODO: fix refund.
+					return Err(Error::<T>::InvalidReplacement.with_weight(2_000_000_000).into());
```

Comment:
The plan is to make the weight of this function be a full block or very high (since it calls into phragmen) but if the prediction was wrong, we take the high weight and do nothing. Needs a refund.